File: skills/vuln-discovery/evals/fixtures/python/sqli_vulnerable.py
"""Vulnerable: SQL injection via string formatting."""
import sqlite3

def get_user(username):
    conn = sqlite3.connect("app.db")
    cursor = conn.cursor()
    query = "SELECT * FROM users WHERE username = '%s'" % username
    cursor.execute(query)
    return cursor.fetchone()

def search_products(term):
    conn = sqlite3.connect("app.db")
    cursor = conn.cursor()
    cursor.execute(f"SELECT * FROM products WHERE name LIKE '%{term}%'")
    return cursor.fetchall()

File: skills/vuln-discovery/evals/fixtures/go/sqli_vulnerable.go
// Vulnerable: SQL injection via Sprintf into db.Query.
package main

import (
	"database/sql"
	"fmt"
	"net/http"

	_ "github.com/lib/pq"
)

var db *sql.DB

func searchHandler(w http.ResponseWriter, r *http.Request) {
	term := r.URL.Query().Get("q")
	query := fmt.Sprintf("SELECT * FROM products WHERE name LIKE '%%%s%%'", term)
	rows, err := db.Query(query)
	if err != nil {
		http.Error(w, err.Error(), 500)
		return
	}
	defer rows.Close()
	fmt.Fprintf(w, "results returned")
}

func userHandler(w http.ResponseWriter, r *http.Request) {
	id := r.URL.Query().Get("id")
	row := db.QueryRow("SELECT email FROM users WHERE id = " + id)
	var email string
	row.Scan(&email)
	fmt.Fprintf(w, email)
}
